src/Raven.Server/Utils/PoolOfThreads.cs

```diff
@@ -232,6 +232,20 @@ private bool DoWork()
                     LongRunningWork.Current = _workIsDone;
                     _action(_state);
                 }
+                catch (OutOfMemoryException e)
```

Comment:
We cannot do this like that. This will ignore OOM exception and e.g. kill the index without killing the server. User will not know that half of his indexes are not working. This needs to be handled internally in all of the actions.